### PR TITLE
Error on conflict between npm shrinkwrap and modern lockfiles

### LINKED ISSUE #430: Check for shrinkwrap + package-lock.json + yarn.lock overlap
If more than one lockfile is included in a repo we should error out. This is currently the behavior with yarn + npm 5, but we should include shrinkwrap as well.

### PR DESCRIPTION
**Pull request by jmorrell**
Fixes #430

This issue is unlikely to occur between `package-lock.json` and `npm-shrinkwrap.json` because npm will automatically use and update the shrinkwrap file, but we should still check against `yarn.lock`. Yarn prints out a warning, but this could be ignored.

hyper 2017-06-13 23-06-11

### CODE REVIEW

File: lib/dependencies.sh [hunk 1]
```diff
@@ -73,7 +73,9 @@ npm_node_modules() {
   if [ -e $build_dir/package.json ]; then
     cd $build_dir
 
-    if [ -e $build_dir/npm-shrinkwrap.json ]; then
+    if [ -e $build_dir/package-lock.json ]; then
+      echo "Installing node modules (package.json + package-lock)"
```
Comment (jmorrell) [inline lib/dependencies.sh]: Went ahead and added this log message to be consistent

**Comment by jmorrell** [comment]
Not sure why travis is failing #willfix

**Comment by jmorrell** [comment]
Definitely want to fail here since we could end up installing wildly different package versions if we use the wrong one. The failure on having two lockfiles (yarn and npm) has come up rarely based on metrics, and I expect this to come up even less,

**Comment by dzuelke** [comment]
CI still failing tho.

File: lib/failure.sh [hunk 2]
```diff
@@ -86,6 +91,29 @@ fail_yarn_and_npm_lockfiles() {
...
+  if $has_modern_lockfile && [ -f "${1:-}/npm-shrinkwrap.json" ]; then
+    mcount "failures.shrinkwrap-lock-file-conflict"
+    header "Build failed"
+    warn "Two different lock files found
+
+       Your application has two lockfiles defined, but only one can be used
+       to install dependencies. Installing dependencies using the wrong lockfile
+       can result in missing packages or subtle bugs in production.
+
+       It's most likely that you recently installed yarn which has it's own
```
Comment (hunterloftis) [inline lib/failure.sh]: Should be "has its own"

File: lib/failure.sh [hunk 3]
```diff
@@ -86,6 +91,29 @@ fail_yarn_and_npm_lockfiles() {
     " https://kb.heroku.com/why-is-my-node-js-build-failing-because-of-conflicting-lock-files
     exit 1
   fi
+
+  if $has_modern_lockfile && [ -f "${1:-}/npm-shrinkwrap.json" ]; then
+    mcount "failures.shrinkwrap-lock-file-conflict"
+    header "Build failed"
+    warn "Two different lock files found
```
Comment (hunterloftis) [inline lib/failure.sh]: "lock files" used here and "lockfiles" below.

**Comment by hone** [comment]
@jmorrell what happens in the npm shrinkwrap + lock case?

**Comment by jmorrell** [comment]
@hone when I tested with both npm defaults to using / updating the shrinkwrap file and doesn't create a `package-lock.json`

**Pull request merged**